src/include/api.h

```diff
@@ -55,6 +55,36 @@
 #define WT_SINGLE_THREAD_CHECK_STOP(s)
 #endif
 
+/*
+ * Helper macros to correctly read and check the API counters. Carefully read the out counter before
+ * the in counter otherwise the out counter can include more API calls than the in and make the
+ * balance negative.
```

Comment:
I'd suggest that the comment should also mention that this requires `in` and `out` counters use sequential consistency (seq-cst) memory ordering.